torchtext/transforms.py

```diff
@@ -189,3 +193,350 @@ def forward(self, input: Any) -> Any:
+                root=_CACHE_DIR
+            )
+        bpe_merge_ranks = GPT2BPETokenizer.load_bpe_vocab(vocab_bpe_local_path)
+        return GPT2BPETokenizer(
+            bpe_encoder,
+            bpe_merge_ranks,
+            GPT2BPETokenizer.load_byte_encoder()
+        )
+
+    @classmethod
+    def load_bpe_encoder(cls, bpe_encoder_path: str) -> Dict[str, int]:
+        if os.path.exists(bpe_encoder_path):
+            local_path = bpe_encoder_path
+        else:
+            local_path = download_from_url(url=bpe_encoder_path, root=_CACHE_DIR)
+        with open(local_path, "r", encoding="utf-8") as f:
+            return json.load(f)
+
+    @classmethod
+    def load_bpe_vocab(cls, bpe_vocab_path: str) -> Dict[str, int]:
+        if os.path.exists(bpe_vocab_path):
+            local_path = bpe_vocab_path
+        else:
+            local_path = download_from_url(url=bpe_vocab_path, root=_CACHE_DIR)
+        with open(local_path, "r", encoding="utf-8") as f:
+            bpe_vocab = f.read()
+
+        return {
+            cls.SEPERATOR.join(merge_pair.split()): i
+            for i, merge_pair in enumerate(bpe_vocab.split("\n")[1:-1])
+        }
+
+    @classmethod
+    def load_byte_encoder(cls) -> Dict[int, str]:
+
+        @lru_cache()
+        def bytes_to_unicode():
+            """
+            Original Source: https://github.com/openai/gpt-2/blob/master/src/encoder.py#L9
+
+            Returns list of utf-8 byte and a corresponding list of unicode strings.
+            The reversible bpe codes work on unicode strings.
+            This means you need a large # of unicode characters in your vocab if you want to avoid UNKs.
+            When you're at something like a 10B token dataset you end up needing around 5K for decent coverage.
+            This is a signficant percentage of your normal, say, 32K bpe vocab.
+            To avoid that, we want lookup tables between utf-8 bytes and unicode strings.
+            And avoids mapping to whitespace/control characters the bpe code barfs on.
+            """
+            bs = (
+                list(range(ord("!"), ord("~") + 1))
+                + list(range(ord("¡"), ord("¬") + 1))
+                + list(range(ord("®"), ord("ÿ") + 1))
+            )
+            cs = bs[:]
+            n = 0
+            for b in range(2 ** 8):
+                if b not in bs:
+                    bs.append(b)
+                    cs.append(2 ** 8 + n)
+                    n += 1
+            cs = [chr(n) for n in cs]
+            return dict(zip(bs, cs))
+
+        return bytes_to_unicode()
+
+    @classmethod
+    def load_bpe_decoder(cls, bpe_encoder: Dict[str, int]) -> Dict[int, str]:
+        bpe_decoder: Dict[int, str] = {}
+        for sub_word, bpe_token_id in bpe_encoder.items():
+            bpe_decoder[bpe_token_id] = sub_word
+        return bpe_decoder
+
+    @classmethod
+    def load_byte_decoder(cls, byte_encoder: Dict[int, str]) -> Dict[str, int]:
```

Comment:
The logic inside of `load_byte_decoder` and `load_bpe_decoder` looks identical. I think they are better extracted to a helper function. (I recommend a plain helper function rather than helper method)

Reply:
By plain helper function do you mean a util function outside the class?

Reply:
I think we plan to refactor tokenizers once this goes out so that each tokenizer be its own module. Once we get to that directory design, we can certainly create helper functions for this class. I further plan to move the logic to C++ operators, so that refactor will also help.  In the meanwhile, I am planning to keep the current methods as it is to prevent duplicate effort.